src/services/comparator.ts

```diff
@@ -0,0 +1,160 @@
+import dotenv from 'dotenv';
+import {
+  ComparatorInput,
+  Destination,
+  ProcessorTransformationRequest,
+  RouterTransformationRequestData,
+} from '../types';
+import ServiceSelector from '../helpers/serviceSelector';
+import tags from '../v0/util/tags';
+import stats from '../util/stats';
+import logger from '../logger';
+import { CommonUtils } from '../util/common';
+import { PlatformError } from '../v0/util/errorTypes';
+const NS_PER_SEC = 1e9;
+
+dotenv.config();
+
+export default class ComparatorService {
+  private static getTestThreshold(destination: Destination) {
+    return destination.DestinationDefinition?.Config['camparisonTestThreshold'] || 0;
+  }
+
+  public static async compareDestinationService(input: ComparatorInput): Promise<void> {
+    const { events, version, destination, requestMetadata, feature } = input;
+
+    try {
+      const primaryServiceName = process.env.COMPARISON_TEST_PRIMARY_SERVICE;
+      const secondaryServiceName = process.env.COMPARISON_TEST_SECONDARY_SERVICE;
+
+      const envThreshold = parseFloat(process.env.COMPARISON_TEST || '0');
+      const destThreshold = this.getTestThreshold(events[0].destination);
+      const compareTestThreshold = envThreshold * destThreshold;
+      if (
```

Comment:
the primary is always called, and secondary is called conditionally based threshold